src/FolderSync.cpp

```diff
@@ -473,16 +473,28 @@ void CFolderSync::SyncFile(const std::wstring& path, const PairData& pt)
             fd.ft = fDataOrig.ftLastWriteTime;
             if (bCopyOnly)
             {
+                bool bCopyFileResult;
                 CCircularLog::Instance()(_T("INFO:    copy file %s to %s"), orig.c_str(), crypt.c_str());
-                if (!CopyFile(orig.c_str(), crypt.c_str(), FALSE))
+                bCopyFileResult = CopyFile(orig.c_str(), crypt.c_str(), FALSE);
+                if (!bCopyFileResult)
                 {
                     std::wstring targetFolder = crypt.substr(0, crypt.find_last_of('\\'));
                     CPathUtils::CreateRecursiveDirectory(targetFolder);
-                    CopyFile(orig.c_str(), crypt.c_str(), FALSE);
+                    bCopyFileResult = CopyFile(orig.c_str(), crypt.c_str(), FALSE);
+                }
+                if (bCopyFileResult && pt.m_ResetOriginalArchAttr)
+                {
+                    // Reset archive attribute on original file
+                    fDataOrig.dwFileAttributes = fDataOrig.dwFileAttributes & (~FILE_ATTRIBUTE_ARCHIVE);
+                    SetFileAttributes(orig.c_str(), fDataOrig.dwFileAttributes);
```

Comment:
you set the file attributes, but that changes the last-write-time of the file, so you have to restore that too.
And of course in other places that SetFileAttributes is used.

Reply:
Will fix shortly. Note that setting FILE_ATTRIBUTE_NOT_CONTENT_INDEXED on the archive will change its timestamp